### ISSUE TITLE
Andromeda - decoy issue

### ISSUE THREAD

**Issue by cdlawless** [compomics/searchgui#105]
Hi, 

I've tried using andromeda via SearchGUI (2.8.6) and provided a SearchGUI formatted fasta database (created by FastaCLI to create the reversed counterparts). However, looking at the .res output it appears that andromeda is also reversing to created decoys and as such is reversing the already reversed sequences. So, the output file shows hits for a sequence as follows (for example):
FCSSMK	19.98396	758.3091312476	-0.003684181	A,A,A,A,A,A,A,A	"78d9e04d771261e9f36afd707cdc0c40:N";"REV__78d9e04d771261e9f36afd707cdc0c40_REVERSED:N"	2

which is showing a hit to both the "real" sequence and the reverse of the FastaCLI reversed sequence (so the original sequence again).

This also happens to "_REVERSED" and "REV__" sequences.

Regards

Craig

**Comment by hbarsnes** [comment]
Hi Craig,

Adding reversed decoys is the default behavior of Andromeda (set via the 'decoy mode=reverse' option). And as the command line options for Andromeda are not very well documented we have not yet figured out if or how it's possible to turns this feature off.

However, when loading Andromeda results (or indeed any of the search engine results) we don't actually load the peptide to protein mappings. The main reason being that the protein inference is not identical across the different search engines, and we therefore have to redo the protein inference for all the peptides loaded into PeptideShaker.

So while the adding of the extra reversed decoys is not optimal (and could perhaps slow down the Andromeda processing), it should not affect the peptide ids. Confirming this would require more testing though, in particular finding out how to turn off the adding of the decoys.

Best regards,
Harald

**Comment by hbarsnes** [comment]
Hi Craig,

I did a very quick test to see which other decoy modes that could be selected, and it turned out that simply changing 'decoy mode=reverse' to 'decoy mode=none' got rid of the Andromeda added decoys.

And the only difference between the two result files was the simplified peptide to protein mapping for the one without the extra decoys added. I only ran the two searches once each, but the one with the 'none' option took 9.5 minutes, while the one with 'reverse' took 15.5 minutes. But we need more tests to see if this trend holds up of course.

Anyway, thanks for reminding us that we needed to look into this. We'll implement the 'none' option as the default in the next release and allow the user to change it via the Andromeda advanced settings.

Best regards,
Harald
